src/DataFramesMeta.jl

```diff
@@ -583,6 +584,59 @@ macro transform(x, args...)
     esc(transform_helper(x, args...))
 end
 
+##############################################################################
+##
+## transform! & @transform!
+##
+##############################################################################
+
+
+function transform!_helper(x, args...)
+
+    t = (fun_to_vec(arg) for arg in args)
+
+    quote
+        $DataFrames.transform!($x, $(t...))
+    end
+end
+
+"""
+    @transform!(d, i...)
+
+Mutate `d` inplace to add additional columns or keys based on keyword arguments and return it. 
+
+### Arguments
+
+* `d` : an `AbstractDataFrame`, or `GroupedDataFrame`
+* `i...` : keyword arguments defining new columns or keys
+
+### Returns
+
+* `::AbstractDataFrame` or `::GroupedDataFrame`
```

Comment:
It will only return an `DataFrame`. 

1. `transform!` on a `SubDataFrame` is not defined
2. We don't have the ability to apply keyword arguments to the macros yet, so we can't pass `ungroup = false` to the call.

Reply:
Got it. Thanks for pointing it out.

In this case, I guess [the return type](https://github.com/JuliaData/DataFramesMeta.jl/blob/e7dde1c0fab9feaecf772dfa64718861c228f5c4/src/DataFramesMeta.jl#L563) for `@transform` is also limited to `DataFrame` only?